services/web/server/src/simcore_service_webserver/projects/projects_handlers.py

```diff
@@ -10,48 +10,57 @@
 
 from aiohttp import web
 from models_library.projects_state import ProjectState
+from pydantic import BaseModel
+from servicelib.aiohttp.requests_validation import (
+    parse_request_path_parameters_as,
+    parse_request_query_parameters_as,
+)
 from servicelib.aiohttp.web_exceptions_extension import HTTPLocked
 from servicelib.json_serialization import json_dumps
+from servicelib.mime import APPLICATION_JSON
 
 from .._meta import api_version_prefix as VTAG
 from ..director_v2_core import DirectorServiceError
-from ..login.decorators import RQT_USERID_KEY, login_required
+from ..login.decorators import login_required
 from ..resource_manager.websocket_manager import PROJECT_ID_KEY, managed_resource
 from ..security_decorators import permission_required
 from . import projects_api
 from .projects_exceptions import ProjectNotFoundError
-from .projects_handlers_crud import routes
+from .projects_handlers_crud import BaseRequestContext, ProjectPathParams
 
 log = logging.getLogger(__name__)
 
+
+routes = web.RouteTableDef()
+
 #
-# Singleton per-user resources https://google.aip.dev/156
+# singleton: Active project
+#  - Singleton per-session resources https://google.aip.dev/156
 #
 
 
-@routes.get(f"/{VTAG}/projects/active")
+class _ProjectActiveParams(BaseModel):
+    client_session_id: str
+
+
+@routes.get(f"/{VTAG}/projects/active", name="get_active_project")
```

Comment:
Q: I see you put the name attribute everywhere. is that mandatory?

Reply:
It is highly advisable but it is also easy to auto assign to the name of the function. Will do that at some point as well